src/common/boundary_condition.cpp

```diff
@@ -123,4 +123,32 @@ void BoundaryCondition::apply(mfem::HypreParMatrix& k_mat_post_elim, mfem::Vecto
   eliminateToRHS(k_mat_post_elim, state.trueVec(), rhs);
 }
 
+const mfem::Coefficient& BoundaryCondition::scalarCoefficient() const
+{
+  SLIC_ERROR_IF(!std::holds_alternative<std::shared_ptr<mfem::Coefficient>>(coef_),
+                "Asing for a scalar coefficient on a BoundaryCondition that contains a vector coefficient.");
+  return *std::get<std::shared_ptr<mfem::Coefficient>>(coef_);
+}
+
+mfem::Coefficient& BoundaryCondition::scalarCoefficient()
+{
+  SLIC_ERROR_IF(!std::holds_alternative<std::shared_ptr<mfem::Coefficient>>(coef_),
```

Comment:
Should this be an assert? It should only come up in solver development.

Reply:
If a user-facing solver ever exposes `GeneralCoef` (the variant) then this might still be useful as an error condition.